ext/pgsql/pgsql.c

```diff
@@ -2147,6 +2148,17 @@ PHP_FUNCTION(pg_trace)
 	}
 	php_stream_auto_cleanup(stream);
 	PQtrace(pgsql, fp);
+	if (trace_mode > 0) {
+#ifdef PQTRACE_REGRESS_MODE
+		if (trace_mode > (PQTRACE_SUPPRESS_TIMESTAMPS|PQTRACE_REGRESS_MODE)) {
```

Comment:
This test is a bit odd, why not `!(trace_mode & (PQTRACE_SUPPRESS_TIMESTAMPS|PQTRACE_REGRESS_MODE))`? This would hide unhandled, smaller flags if they didn't start at `0`.

Reply:
Does this also need a `php_stream_close(stream);` as above?